src/Clock.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+
+function getCurrentClockTime() {
+  return new Date().toUTCString().slice(-12, -4);
+}
+
+type ClockProps = {
+  name: string
+};
+
+export class Clock extends React.Component<ClockProps> {
+  state = {
+    time: getCurrentClockTime(),
+    timeUpdaterId: -1,
+  };
+
+  componentDidMount() {
+    this.setState({
+      timeUpdaterId: window.setInterval(() => {
+        const time = getCurrentClockTime();
+
+        // eslint-disable-next-line no-console
+        console.info(time);
+        this.setState({
+          time,
+        });
+      }, 1000),
+    });
+  }
+
+  componentDidUpdate(prevProps: Readonly<ClockProps>) {
+    const oldName = prevProps.name;
+    const newName = this.props.name;
+
+    if (oldName !== newName) {
```

Comment:
There is no need for declaring two variables for the sake of renaming values, just
```ts
if (this.props.name !== prevProps.name) {
```

Reply:
I destructured props.